src/stac_utils/google.py

```diff
@@ -71,7 +80,8 @@ def run_query(
         )
         client = bigquery.Client(credentials=credentials)
 
-    job = client.query(sql).result()
+    retry_policy = Retry(predicate=if_exception_type(*RETRY_EXCEPTIONS))
```

Comment:
I think we should have a timeout as well as a list of exceptions. Otherwise we would get into an infinite loop if the error keeps coming.

Reply:
There's a default time out of 120 seconds total. It retries with backoff and then ultimately fails when it hits that limit.

Reply:
Cool. That works then. Just didn't want to risk an infinite loop